src/main/src/main/java/org/geoserver/catalog/impl/ResourceInfoImpl.java

```diff
@@ -241,30 +241,55 @@ public ReferencedEnvelope boundingBox() throws Exception {
         CoordinateReferenceSystem nativeCRS = getNativeCRS();
         ProjectionPolicy php = getProjectionPolicy();
 
-        ReferencedEnvelope nativeBox = this.nativeBoundingBox;
+        ReferencedEnvelope nativeBox = getNativeBoundingBox();
         if (nativeBox == null) {
             // back project from lat lon
             try {
-                nativeBox = getLatLonBoundingBox().transform(declaredCRS, true);
+                if (declaredCRS != null) {
+                    nativeBox = getLatLonBoundingBox().transform(declaredCRS, true);
+                } else {
+                    LOGGER.log(
+                            Level.WARNING,
+                            "Failed to derive native bbox, there is no declared CRS provided");
+                    return null;
+                }
             } catch (Exception e) {
                 LOGGER.log(Level.WARNING, "Failed to derive native bbox from declared one", e);
                 return null;
             }
+        } else if (nativeBox.getCoordinateReferenceSystem() == null) {
+            if (nativeCRS != null) {
+                // Ensure provided nativeBox using nativeCRS
+                nativeBox = ReferencedEnvelope.create(nativeBox, nativeCRS);
+            }
+        } else if (nativeCRS != null
+                && !CRS.equalsIgnoreMetadata(nativeBox.getCoordinateReferenceSystem(), nativeCRS)) {
+            LOGGER.log(Level.FINE, "The native bounding box srs does not match native crs");
+            nativeBox = ReferencedEnvelope.create(nativeBox, nativeCRS);
```

Comment:
I have doubts about this bit... those two are initially set up together, but then one can request to recompute the bbox through the REST API I believe? And then the bbox might end up being more up-to-date than the native CRS?
I haven't checked in detail.

Reply:
The UI only allows changing of nativeCRS so I have to trust that one :)

Reply:
The UI does not, at least not in core GeoServer. Here is a screenshot from ArcSample: 

![image](https://github.com/geoserver/geoserver/assets/325433/77c1f9c2-39cd-48a2-a78c-babee1bbe38f)

Try typing into the "native srs" field... it won't work. Only declared SRS can be modified.
The UI used to have a different color for the native SRS making it clear that it was not editable, it might have broken with the CSS update?

Reply:
@MichelGabriel see above!

Reply:
> I have doubts about this bit... those two are initially set up together, but then one can request to recompute the bbox through the REST API I believe? And then the bbox might end up being more up-to-date than the native CRS?

You are correct:
```
        if (fieldsToCalculate.contains("nativebbox")) {
            CatalogBuilder builder = new CatalogBuilder(catalog);
            try {
                message.setNativeBoundingBox(builder.getNativeBounds(message));
            } catch (IOException e) {
                String errorMessage = "Error while calculating native bounds for layer: " + message;
                throw new RestException(errorMessage, HttpStatus.INTERNAL_SERVER_ERROR, e);
            }
        }
```

No attempt to update the native crs to match newly calculated native bounds.